ci/do_ci.sh

```diff
@@ -89,8 +89,8 @@ if [[ "$1" == "bazel.release" ]]; then
     # https://github.com/envoyproxy/envoy/pull/5611.
     # TODO(akonradi): use --local_cpu_resources flag once Bazel has a release
     # after 0.21.
-    [ -z "$CIRCLECI" ] || export BAZEL_BUILD_OPTIONS="${BAZEL_BUILD_OPTIONS} --local_resources=12288,6,1"
-    [ -z "$CIRCLECI" ] || export BAZEL_TEST_OPTIONS="${BAZEL_TEST_OPTIONS} --local_resources=12288,6,1"
+    [ -z "$CIRCLECI" ] || export BAZEL_BUILD_OPTIONS="${BAZEL_BUILD_OPTIONS} --local_resources=12288,5,1"
```

Comment:
I'm not sure how much of a speedup it would be, but I'm still pretty sure you could build `//source` with full CPUs, and then proceed to build the tests with less.